pkg/cloudprovider/openstack.go

```diff
@@ -495,37 +492,14 @@ func (o *OpenStack) getNeutronPortNodeEgressIPConfiguration(p neutronports.Port)
 			IPv6: ipv6,
 		},
 		Capacity: capacity{
-			IPv4: ipv4Cap - ipv4UsedIPs,
-			IPv6: ipv6Cap - ipv6UsedIPs,
+			IP: openstackMaxCapacity - numAddressPairs,
 		},
 	}, nil
 }
 
-// getIPsOnPort returns the number of unique IP addresses on the given port.
-// Those include both IPs in the list of FixedIP addresses and in the list of
-// allowed_address_pairs. This method is a helper for getNeutronPortNodeEgressIPConfiguration
-// and as such does not test if multiple networks of the same address family are assigned to
-// the port, given that the calling method will already have done this.
-func (o *OpenStack) getIPsOnPort(p neutronports.Port) (int, int) {
-	ipv4UsedIPs := make(map[string]struct{})
-	ipv6UsedIPs := make(map[string]struct{})
-
-	for _, ip := range p.FixedIPs {
-		if utilnet.IsIPv4(net.ParseIP(ip.IPAddress)) {
-			ipv4UsedIPs[ip.IPAddress] = struct{}{}
-		} else {
-			ipv6UsedIPs[ip.IPAddress] = struct{}{}
-		}
-	}
-	for _, ip := range p.AllowedAddressPairs {
-		if utilnet.IsIPv4(net.ParseIP(ip.IPAddress)) {
-			ipv4UsedIPs[ip.IPAddress] = struct{}{}
-		} else {
-			ipv6UsedIPs[ip.IPAddress] = struct{}{}
-		}
-	}
-
-	return len(ipv4UsedIPs), len(ipv6UsedIPs)
+// countAllowedAddressPairsOnPort returns the number of allowed_address_pairs on the port.
+func (o *OpenStack) countAllowedAddressPairsOnPort(p neutronports.Port) int {
```

Comment:
NIT: instead of having that function can we just use `len(p.AllowedAddressPairs)` instead?

Reply:
+1